controllers/managedmcg_controller.go

```diff
@@ -531,3 +510,50 @@ func (r *ManagedMCGReconciler) lookupImages() error {
 	}
 	return nil
 }
+
+func (r *ManagedMCGReconciler) get(obj client.Object) error {
+	key := client.ObjectKeyFromObject(obj)
+	return r.Client.Get(r.ctx, key, obj)
+}
+
+func (r *ManagedMCGReconciler) list(obj client.ObjectList) error {
+	listOptions := client.InNamespace(r.namespace)
+	return r.Client.List(r.ctx, obj, listOptions)
+}
+
+func (r *ManagedMCGReconciler) update(obj client.Object) error {
+	return r.Client.Update(r.ctx, obj)
+}
+
+func (r *ManagedMCGReconciler) delete(obj client.Object) error {
+	if err := r.Client.Delete(r.ctx, obj); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
+
+func (r *ManagedMCGReconciler) own(resource metav1.Object) error {
+	// Ensure ManagedMCG ownership on a resource
+	if err := ctrl.SetControllerReference(r.managedMCG, resource, r.Scheme); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ManagedMCGReconciler) getCSVByPrefix(name string) (*opv1a1.ClusterServiceVersion, error) {
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err := r.list(&csvList); err != nil {
+		return nil, fmt.Errorf("unable to list csv resources: %v", err)
+	}
+	var csv *opv1a1.ClusterServiceVersion = nil
+	for _, candidate := range csvList.Items {
+		if strings.HasPrefix(candidate.Name, name) {
+			csv = &candidate
+			break
+		}
+	}
+	if csv == nil {
+		return nil, fmt.Errorf("unable to get csv resources for %s ", name)
+	}
+	return csv, nil
+}
```

Comment:
Can we move these helper functions to `/utils`?

Reply:
now the util calss is out side the `controller` package because of that we cannot access `ManagedMCGReconciler` in util package

Reply:
Can we send that as a function parameter instead of a receiver? Following this pattern will allow us to trim down help function population in the core logic.